user-guide/Advanced_Modules/DOM/DOM_examples/DOM_BulkProcessing_Example.md

```diff
@@ -0,0 +1,135 @@
+
+Like in the previous example, some `DomInstances` need to be updated:
+
+```csharp
+// Update the DomInstances.
+
+// Save them to the DB.
+var updateResult = domHelper.DomInstances.CreateOrUpdate(domInstances);
+
+// Check if some of the updates are not successful.
+if (updateResult.HasFailures())
+{
+  // Log the number of DomInstances that was not updated.
+  // Also log the `TraceData` for all failing `DomInstances`. The `TraceData` contains all errors and warnings.
+  Log($"Could not perform the update for {updateResult.UnsuccessfulIds.Count} items: {updateResult.GetTraceData()}");
+}
+
+// Log what items were successfully removed.
+Log($"Could perform the update successfully for {updateResult.SuccessfulItems.Count} items");
+```
+
+The result returned by `CreateOrUpdate` contains `SuccessfulItems` and `SuccessfulIds` to reuse or check the `DomInstances` that are successfully created or updated. The IDs of the items that did not succeed are available in `UnsuccessfulIds`.
+
+In addition to a summarized logging using `GetTraceData()`, `TraceDataPerItem` can be used to check for errors and warnings per `DomInstanceId`.
+
+## Remove multiple DomInstances
+
+In this example, some `DomInstances` get filtered out to be removed.
+
+```csharp
+// Retrieve the DomInstances that need to get deleted.
+var domInstances = domHelper.DomInstances.Read(filter);
+
+// Remove them from the DB.
+var deleteResult = domHelper.DomInstances.Delete(domInstances);
+```
+
+Similar to the [previous example](xref:DOM_BulkProcessing_Example#checking-issues), it is possible to get `UnsuccessfulIds` and `TraceDataPerItem` and to call `HasFailures()` and `GetTraceData()` using the `deleteResult` to log any failures that occurred. However, the `deleteResult` only contains the `SuccessfulIds`.
+
+## Unexpected issue
+
+In case a `CreateOrUpdate` or a `Delete` call are unsuccessful, for instance because of a database issue, by default a `CrudFailedException` will be thrown. Details of the issue will be available in the `TraceData` of that exception.
+
+In the following example, some `DomInstances` need to be deleted:
+
+```csharp
+try
+{
+  // Remove them from the DB.
+  var deleteResult = domHelper.DomInstances.Delete(domInstances);
+
+  // Log what items are successfully removed.
+  Log($"Successfully removed {deleteResult.SuccessfulIds.Count} items");
+}
+catch (CrudFailedException ex)
+{
+  // Log the issue that occurs.
+  Log($"Cannot perform the removal, because of the following issue: {ex.TraceData}");
+}
+```
+
+It is also possible to disable this and check if something went wrong yourself by requesting the `TraceData` object of the last call.
+
+```csharp
+// Disable the exceptions.
+domHelper.DomInstances.ThrowExceptionsOnErrorData = false;
+
+// Remove them from the DB.
+var deleteResult = domHelper.DomInstances.DomInstances.Delete(domInstances);
+
+// Check if a result is available.
```

Comment:
Can this block be removed (lines 120-126)? (discussed with TGH that it would be clearer to focus on calling the GetTraceDataLastCall like we do for other methods)

Reply:
I have my doubts about this one. Primary source of checking what when right or wrong should be *deleteResult*. It should never be null, but because *ThrowExceptionsOnErrorData*, it could be in some rare cases.
Maybe the following would make that clearer:

```cs
if (deleteResult == null)
{
  // Get the TraceData and check if the last call succeeded.
  var traceData = domHelper.DomInstances.GetTraceDataLastCall();

  // Log the issue that occurs.
  Log($"An unexpected issue occurred during the removal, because of the following issue: {traceData}");
  return;
}

// Log what items are successfully removed.
Log($"Successfully removed {deleteResult.SuccessfulIds.Count} items");
```

Reply:
As discussed I've implemented the above suggestion and also included the *HasFailures* path.

fyi. I didn't add that in the example initially, since currently on DomInstance level there's no proxy that will cause an error during a delete. But such a proxy could get added in the future.